Add Base HTTP Server header files

jajanet opened the pull request:
This is a PR for the HTTP server base class, which is a lightweight server that other features can use for testing over HTTP protocol. zPages HTTP is currently derived using this to serve files and communicate data. This is not meant for production use, as it doesn't implement HTTPS.

Furthermore, future patches will add additional thread safety and the directory storing these files will be less nested

maxgolov commented:
I think we should add it to `ext/src/http/server`?   `ext/include/opentelemetry/ext/http/server/HttpServer.h` looks a bit too deep, although I do not have a strong preference. Since it's header-only - this may be OK location to use too.

Re. safety features - I can add a few lockguard checks in parts of code that accept a new HTTP request, and/or for the case when the same box is being accessed by multiple concurrent clients. There is usually a practical limit to the number of concurrent HTTP requests from one client machine, so the likelihood of a race condition is rather small. I can patch this, you do not have to worry about it.

I will approve, use your best judgement to decide on location.

jajanet commented:
> I think we should add it to `ext/src/http/server`? `ext/include/opentelemetry/ext/http/server/HttpServer.h` looks a bit too deep, although I do not have a strong preference. Since it's header-only - this may be OK location to use too.

@maxgolov I agree 100%, I think after all the zPages stuff is approved and merged we will make a future PR to move the zPages and HTTP server to different folders, i.e. `ext/zpages/*` and `ext/http/server/*` instead of `ext/include/zpages` `ext/include/http/server` and `ext/src/zpages`.

Also, I added a file server that comes from your example; let me know if it looks okay. It can be used as a base for other HTTP servers, or serve as an example on how to derive an HTTP server using the base http_server.

I also ran the formatter and changed some small cosmetic things (file names) to be consistent with the rest of the repo and to try to pass the CI checks (for some reason they fail, and I think others are having similar issues)

reyang commented:
The CI is failing and it seems to be a transient error.
